code/modules/mob/mob.dm

```diff
@@ -85,7 +85,7 @@ var/next_mob_id = 0
 	var/list/atom_viewers = list()
 	for(var/atom/movable/A in view(src))
 		atom_viewers |= recursive_hear_check(A)
-	atom_viewers |= src
+	atom_viewers |= viewers(src)
```

Comment:
This would mean the src is sometimes not added to the list. You need to keep the `atom_viewers |= src` because the src must always.

You also need to add your line to  /atom/proc/visible_message().